eng/common/tsp-client/README.md

```diff
@@ -0,0 +1,101 @@
+# TypeSpec Client Generator CLI
+
+This directory contains npm package definitions for `@azure-tools/typespec-client-generator-cli` (tsp-client) with pinned versions to ensure reproducible builds across environments and enhance security by preventing supply chain attacks.
+
+## Files
+
+- **`package.json`** - npm package definition with pinned tsp-client version
+- **`package-lock.json`** - Lock file ensuring exact dependency versions
+
+## Prerequisites
+
+- **Node.js** (with npm) - Required to install and run tsp-client
+
+## Installation
+
+### Install dependencies
+
+```bash
+# Navigate to this directory
+cd eng/common/tsp-client
+
+# Install dependencies (recommended for CI/security)
+npm i --ignore-scripts --no-audit
+```
+
+The `--ignore-scripts --no-audit` flags provide:
+
+- **Security**: Prevents potentially malicious install scripts from running
+- **Performance**: Skips vulnerability audit for faster installs
+- **Reliability**: Reduces potential points of failure during installation
+
+## Usage
+
+After installation, you can run tsp-client using one of these approaches:
+
+### Option 1: Change directory approach (recommended)
+
+```bash
+cd eng/common/tsp-client
```

Comment:
Does it require changing the current working directory? @mikeharder is there an option or some way to run this without switching working directories?

Reply:
Is that the prefix option below?

Reply:
According to this, they recommend changing the working directory, rather than using the `--prefix` flag:

https://github.com/npm/cli/issues/1368#issuecomment-1241076700

I'll keep looking, but unless I can find something, we'll need to keep the guidance to use the working directory.

Reply:
Per my comment, I don't think we should recommend `--prefix` at all.  Only the option to change the working directory.